test/y2storage/callbacks/probe_test.rb

```diff
@@ -29,4 +29,83 @@
     include_examples "general #error examples"
     include_examples "default #error true examples"
   end
+
+  describe "#begin" do
+    it "clears again flag" do
+      subject.begin
+      expect(subject.again?).to be false
+    end
+  end
+
+  describe "#again?" do
+    it "is unset after calling begin" do
+      subject.begin
+      expect(subject.again?).to be false
+    end
+  end
+
+  describe "#missing_command" do
+    it "just displays the error and the details if used_features is 0" do
+      expect(Yast::Report).to receive(:yesno_popup) do |message, options|
+        expect(message).to include "the message"
+        expect(options[:details]).to eq "the what"
+      end
+      subject.missing_command("the message", "the what", "the command", 0)
+    end
+
+    before do
+      allow(Yast::Mode).to receive(:installation).and_return false
+      allow(Y2Storage::PackageHandler).to receive(:new).and_return package_handler
+    end
+
+    let(:package_handler) do
+      double("PackageHandler", add_feature_packages: packages, compact: packages)
+    end
+
+    let(:packages) { ["btrfsprogs", "e2fsprogs"] }
```

Comment:
order here is a bit confusing, as it is used even for first `it` statement. So usual order is `let`, then `before`, `after`, `around` and then list of `it` statements.

Try yourself in simple rspec:
```
describe "A" do
  it "says B" do
    puts "B"
  end

  before do
    puts "A"
  end
end
```

Reply:
At least is seems common to first define ```before``` and then ```let``` inside ```it```.

Reply:
I think is a matter of taste. I also like to put `let`s before `before`, but it's true that there are so many tests in storage-ng where `let` s are placed after `before`.

Reply:
I do not have strong opinion about `let` and `before` order, but having `it` before them is confusing for readers as it affect also code in `it`.

Reply:
Ah! I overlooked that `it` placed before `before`.

I agree with @jreidinger again

Reply:
Yes, a matter of taste. What I usually do:

~~~
context "when sky is grey" do
  let(:sky) { "grey" } # this let is setting the condition of this context

  before do
    allow(foo).to receive(:bar).and_return(foobar)
  end

  let(:foobar) { whatever } # this let is defining something used previously in the *before* block
~~~

But this is only my own convention.